### ISSUE TITLE
Demo not working properly

### ISSUE THREAD

**Issue by Apollorion** [github/branch-deploy#49]
I made minor changes to the demo and cannot get it to work. Added `issues: write` to the permissions and `environment: "test"`.

```yaml
name: "branch deploy demo"
on:
  issue_comment:
    types: [created]
permissions:
  pull-requests: write
  deployments: write
  contents: write
  issues: write

jobs:
  demo:
    if: ${{ github.event.issue.pull_request }}
    runs-on: ubuntu-latest
    steps:
      - uses: github/branch-deploy@v2.2.0
        id: branch-deploy
        with:
          trigger: ".deploy"
          environment: "test"
      - uses: actions/checkout@v3
        with:
          ref: ${{ steps.branch-deploy.outputs.ref }}
      - name: fake noop deploy
        if: ${{ steps.branch-deploy.outputs.continue == 'true' && steps.branch-deploy.outputs.noop == 'true' }}
        run: echo "I am doing a fake noop deploy"
      - name: fake regular deploy
        if: ${{ steps.branch-deploy.outputs.continue == 'true' && steps.branch-deploy.outputs.noop != 'true' }}
        run: echo "I am doing a fake regular deploy"

```

Getting this error:
```
GraphqlResponseError: Request failed due to following response errors:
 - Resource not accessible by integration
    at /home/runner/work/_actions/github/branch-deploy/v2.2.0/webpack:/branch-deploy-action/node_modules/@octokit/graphql/dist-node/index.js:81:1
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at prechecks (/home/runner/work/_actions/github/branch-deploy/v2.2.0/webpack:/branch-deploy-action/src/functions/prechecks.js:166:1)
    at run (/home/runner/work/_actions/github/branch-deploy/v2.2.0/webpack:/branch-deploy-action/src/main.js:264:1)
```

**Comment by Apollorion** [comment]
Found if I set the permission to `write-all` it works  ¯\_(ツ)_/¯

**Comment by GrantBirki** [comment]
@Apollorion It looks like your fix was to do the following?

```yaml
permissions: write-all
```

I have noticed in a few repos where I have set this up for that to be the case as well.

Anytime I get `Resource not accessible by integration` it is a matter of tweaking, removing, or altering the `permissions` block (or removing it altogether) in order to get things working.

I aim to have this fixed and smoothed out in a future release.

Closing for now as you got it working :smiley:
